### DISCUSSION TITLE
electron.net use for remote transport

### DISCUSSION THREAD

**Discussion by psiservices-ccounterman** [megahertz/electron-log#351] [Ideas]
I'm having problems with people running my electron app behind a proxy.  I've fixed some issues by using 
https://www.electronjs.org/docs/latest/api/net which "is similar to the [HTTP](https://nodejs.org/api/http.html) and [HTTPS](https://nodejs.org/api/https.html) modules of Node.js but uses Chromium's native networking library instead of the Node.js implementation, offering better support for web proxies."
I'm fixing more, but realized that it is affecting the logging as well.  
src/transports/remote.js uses http and https.
Before I do anything difficult, is there a currently-supported way to swap in the electron net API or substitute a different remote transport?
Is there a reason electron-log shouldn't use electron.net as the norm?

**Comment by megahertz** [comment]
Currently, the it's not possible to change a way how a request is sent. There are two workarounds:
- Replace the whole remote transport, hopefully that's pretty easy
- Create an Agent which uses `electron.net` and assign it to `requestOptions.agent`

**Reply by psiservices-ccounterman** [reply]
Thanks.
I've been looking at node's http.Agent and agent proxy implementations, but I haven't been able to see how to hook in
electron's net.request in an Agent.  It would be the most useful thing, but I'm not getting how it'd be done.

**Reply by megahertz** [reply]
I'm not familiar with it, but I think overriding the whole remote transport is simpler. Anyway, sometimes it's useful to configure log sending, so I'm going to implement that in the future https://github.com/megahertz/electron-log/issues/352
